### PR TITLE
Add content creator

### PR DESCRIPTION
**Pull request by avidan-H**
changes made in `content_creator.py`, `unifier.py`, and `core.py`

### CODE REVIEW

File: demisto_sdk/common/tools.py [hunk 1]
```diff
@@ -134,6 +134,28 @@ def filter_packagify_changes(modified_files, added_files, removed_files, tag='ma
     return modified_files, updated_added_files, removed_files
 
 
+def get_child_directories(directory):
```
Comment (liorblob) [inline demisto_sdk/common/tools.py]: isn't there something built in for this?

File: demisto_sdk/common/tools.py [hunk 2]
```diff
@@ -134,6 +134,28 @@ def filter_packagify_changes(modified_files, added_files, removed_files, tag='ma
...
+    if not os.path.isdir(directory):
+        return []
+    child_directories = [
+        os.path.join(directory, path) for
+        path in os.listdir(directory) if os.path.isdir(os.path.join(directory, path))
+    ]
+    return child_directories
+
+
+def get_child_files(directory):
```
Comment (liorblob) [inline demisto_sdk/common/tools.py]: isn't there something built in for this?

File: demisto_sdk/yaml_tools/content_creator.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
+import os
+import glob
+import shutil
+import zipfile
+import io
+import yaml
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: all of these are built in right?

File: demisto_sdk/yaml_tools/content_creator.py [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+    WIDGETS_DIR, SCRIPTS_DIR, INCIDENT_FIELDS_DIR, CLASSIFIERS_DIR, LAYOUTS_DIR, CONNECTIONS_DIR, \
+    BETA_INTEGRATIONS_DIR, INDICATOR_FIELDS_DIR, INCIDENT_TYPES_DIR, TEST_PLAYBOOKS_DIR, PACKS_DIR, DIR_TO_PREFIX
+from ..common.tools import get_child_directories, get_child_files, print_warning
+from .unifier import Unifier
+
+
+class ContentCreator:
+
+    def __init__(self, artifacts_path: str, content_bundle_path='',
+                 test_bundle_path='', packs_bundle_path='', preserve_bundles=False):
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: do we need all these arguments as we're not sending them from the core?

Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: Add some doc strings

File: demisto_sdk/yaml_tools/content_creator.py [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+        if not dest_file_name.startswith('playbook-'):
+            new_name = '{}{}'.format('playbook-', dest_file_name)
+            out_path = os.path.join(dest_dir_path, new_name)
+        shutil.copyfile(path, out_path)
+
+    def copy_content_yml(self, path, out_path, yml_info):
+        parent_dir_name = os.path.basename(os.path.dirname(path))
+        if parent_dir_name in DIR_TO_PREFIX and not os.path.basename(path).startswith('playbook-'):
+            script_obj = yml_info
+            if parent_dir_name != 'Scripts':
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: Use CONSTS

File: demisto_sdk/yaml_tools/content_creator.py [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+            for root, _, files in os.walk(directory):
+                for file_name in files:
+                    zipf.write(os.path.join(root, file_name), file_name)
+            zipf.close()
+
+    def copy_playbook_yml(self, path, out_path, *args):
+        '''Add "playbook-" prefix to playbook file's copy destination filename if it wasn't already present'''
+        dest_dir_path = os.path.dirname(out_path)
+        dest_file_name = os.path.basename(out_path)
+        if not dest_file_name.startswith('playbook-'):
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: Const?

File: demisto_sdk/yaml_tools/content_creator.py [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+            unification_tool = Unifier(package, package_dir_name, dest_dir)
+            if any(package_to_skip in package for package_to_skip in self.packages_to_skip):
+                # there are some packages that we don't want to include in the content zip
+                # for example HelloWorld integration
+                unification_tool = Unifier(package, package_dir_name, skip_dest_dir)
+                print('skipping {}'.format(package))
+            unification_tool.merge_script_package_to_yml()
+
+    def add_tools_to_bundle(self, bundle):
+        for directory in glob.glob(os.path.join('Tools', '*')):
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: Const?

File: demisto_sdk/yaml_tools/content_creator.py [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+            self.copy_packs_content_to_old_bundles(packs)
+
+            # handle copying packs content to packs_bundle for zipping to `content_packs.zip`
+            self.copy_packs_content_to_packs_bundle(packs)
+
+            print('Copying content descriptor to content and test bundles')
+            for bundle_dir in [self.content_bundle, self.test_bundle]:
+                shutil.copyfile('content-descriptor.json', os.path.join(bundle_dir, 'content-descriptor.json'))
+            
+            if os.path.exists('./Documentation/doc-CommonServer.json'):
```
Comment (liorblob) [inline demisto_sdk/yaml_tools/content_creator.py]: Need to test this in the content repo

File: demisto_sdk/yaml_tools/content_creator.py (hunk 4 above)
Comment (avidan-H) [reply demisto_sdk/yaml_tools/content_creator.py]: no, but it does allow you if you wanted to import `ContentCreator` class for some reason in your code somewhere and wanted to set the paths for the bundles then you could do that

File: demisto_sdk/common/tools.py (hunk 1 above)
Comment (avidan-H) [reply demisto_sdk/common/tools.py]: you tell me. is there?

File: demisto_sdk/common/tools.py (hunk 2 above)
Comment (avidan-H) [reply demisto_sdk/common/tools.py]: same as above

File: demisto_sdk/yaml_tools/content_creator.py (hunk 3 above)
Comment (avidan-H) [reply demisto_sdk/yaml_tools/content_creator.py]: what do you mean?

**Pull request merged**